component/web/security/src/main/java/org/exoplatform/web/login/externalRegistration/ExternalRegistrationHandler.java

```diff
@@ -217,6 +217,7 @@ public boolean execute(ControllerContext context) throws Exception {
                         return dispatch("/externalRegistration/jsp/reset_password.jsp", servletContext, req, res);
                     }
                     res.sendRedirect("/" + currentPortalContainerName + LOGIN + "?email=" + email);
+                    remindPasswordTokenService.deleteTokensByUsernameAndType(email,remindPasswordTokenService.EXTERNAL_REGISTRATION_TOKEN);
```

Comment:
If there is an error in the user creation, I'm afraid that token are dropped.
Can you delete token only if we send externalConfirmationAccount email